scheduler/filter/slots.go

```diff
@@ -0,0 +1,48 @@
+package filter
+
+import (
+	"errors"
+	"github.com/docker/swarm/cluster"
+	"github.com/docker/swarm/scheduler/node"
+	"strconv"
+)
+
+var (
+	// ErrNoNodeWithFreeSlotsAvailable is exported
+	ErrNoNodeWithFreeSlotsAvailable = errors.New("No node with enough open slots available in the cluster")
+)
+
+//SlotsFilter only schedules containers with open slots.
+type SlotsFilter struct {
+}
+
+// Name returns the name of the filter
+func (f *SlotsFilter) Name() string {
+	return "containerslots"
+}
+
+// Filter is exported
+func (f *SlotsFilter) Filter(_ *cluster.ContainerConfig, nodes []*node.Node, _ bool) ([]*node.Node, error) {
+	result := []*node.Node{}
+
+	for _, node := range nodes {
+
+		if slotsString, ok := node.Labels["containerslots"]; ok {
+			slots, err := strconv.Atoi(slotsString) //if err => cannot cast to int, so no candidate for us
```

Comment:
I'm open to any ideas.... but i don't know, what i should do alternatively...
Return an error, to prevent starting a container at all?